libs/community/langchain_google_community/bq_storage_vectorstores/_base.py

```diff
@@ -0,0 +1,520 @@
+from __future__ import annotations
+
+import asyncio
+import uuid
+from datetime import timedelta
+from functools import partial
+from importlib.util import find_spec
+from threading import Lock
+from typing import Any, Dict, List, Optional, Tuple, Type, Union
+
+import numpy as np
+from google.cloud.exceptions import NotFound
+from langchain_community.vectorstores.utils import maximal_marginal_relevance
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.vectorstores import VectorStore
+from pydantic import BaseModel
+
+from langchain_google_community._utils import get_client_info
+from langchain_google_community.bq_storage_vectorstores.utils import (
+    validate_column_in_bq_schema,
+)
+
+_vector_table_lock = Lock()  # process-wide BigQueryVectorSearch table lock
+
+# Constants for index creation
+MIN_INDEX_ROWS = 5
+INDEX_CHECK_INTERVAL = timedelta(seconds=60)
+USER_AGENT_PREFIX = "FeatureStore"
+
+
+class BaseBigQueryVectorStore(VectorStore, BaseModel):
+    embedding: Any
+    project_id: str
+    dataset_name: str
+    table_name: str
+    location: str
+    content_field: str = "content"
+    text_embedding_field: str = "text_embedding"
+    doc_id_field: str = "doc_id"
+    credentials: Optional[Any] = None
+    embedding_dimension: Optional[int] = None
+    _extra_fields: Union[Dict[str, str], None] = None
+    _table_schema: Any = None
+
+    def sync(self) -> None:
+        raise NotImplementedError()
+
+    def get_documents(
```

Comment:
let's mark this class an abstract one, please, and mark method as an abstract one too.